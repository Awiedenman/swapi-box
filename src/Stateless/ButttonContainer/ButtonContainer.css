.button-container {
  position: absolute;
  top: 0;
  background-color: black;
  border-bottom: yellow 2px solid;
  height: 200px;
  width: 100%;
  z-index: 1;
  box-sizing: border-box;
}

.title {
  color: #ffff00;
  margin: 5px 0 10px;
  font-size: 100px;
}