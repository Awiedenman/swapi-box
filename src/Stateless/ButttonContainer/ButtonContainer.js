import React from 'react';
import './ButtonContainer.css';
import Button from '../Button/Button';
import Favorites from '../../Stateless/Favorites/Favorites';
import PropTypes from 'prop-types';

const ButtonContainer = ( { setData, count, showFavorites } ) => {
  console.log( setData )

   return (
     <div className="button-container">
     < h1 className='title'> swapi-box </h1>
      < Favorites
          count={ count }
          showFavorites={ showFavorites }/>
       < Button 
          setData={ setData }
          name='people' />
       < Button 
          name='vehicles'
          setData = { setData } />
       < Button 
          name='planets' 
          setData = { setData } />
     </div>
   )
}

ButtonContainer.propTypes = {
  setData: PropTypes.func.isRequired,
  count: PropTypes.number.isRequired,
  showFavorites: PropTypes.func.isRequired
}

export default ButtonContainer;