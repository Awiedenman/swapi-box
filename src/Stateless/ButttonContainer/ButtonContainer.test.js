import React from 'react';
import ButtonContainer from "./ButtonContainer";
import { shallow } from 'enzyme';


describe( 'ButtonContainer', () => {
  const mockCount = 1;
  const MockShowFavorite = jest.fn();
  const wrapper = shallow(
    < ButtonContainer
      count = { mockCount }
      showFavorites = { MockShowFavorite }
    />
  );
  it( 'should match the snapshot of the buttons when the page loads', () => {
    expect( wrapper ).toMatchSnapshot();
  });
});