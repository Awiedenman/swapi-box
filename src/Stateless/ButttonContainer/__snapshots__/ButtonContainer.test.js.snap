// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should match the snapshot of the buttons when the page loads 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonContainer
    count={1}
    showFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
           swapi-box 
</h1>,
        <Favorites
          count={1}
          showFavorites={[Function]}
/>,
        <Button
          name="people"
          setData={undefined}
/>,
        <Button
          name="vehicles"
          setData={undefined}
/>,
        <Button
          name="planets"
          setData={undefined}
/>,
      ],
      "className": "button-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " swapi-box ",
          "className": "title",
        },
        "ref": null,
        "rendered": " swapi-box ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "count": 1,
          "showFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "people",
          "setData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "vehicles",
          "setData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "planets",
          "setData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
             swapi-box 
</h1>,
          <Favorites
            count={1}
            showFavorites={[Function]}
/>,
          <Button
            name="people"
            setData={undefined}
/>,
          <Button
            name="vehicles"
            setData={undefined}
/>,
          <Button
            name="planets"
            setData={undefined}
/>,
        ],
        "className": "button-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " swapi-box ",
            "className": "title",
          },
          "ref": null,
          "rendered": " swapi-box ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": 1,
            "showFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "people",
            "setData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "vehicles",
            "setData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "planets",
            "setData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
